reactioncommerce/reaction-next-starterkit
# Enable/Disable Single-page app routing

Resolves #417 
Impact: **minor**
Type: **feature**

## Issue
Description of the issue this PR is solving, why it's happening, and how to reproduce it. This may differ from the original ticket as you now have more information at your disposal.

## Solution

- Add an .env var to enable SPA routing
- Extend next-routes to allow for full page loads
- Update the link component to allow for full page loads
- Update PDP to force it into single-page app mode for variant selection
- Update RoutingStore to for it into single-page app mode when updating the search query (used for the grid filtering)

## Breaking changes

`ENABLE_SPA_LINKS=true` should be placed in your `.env` file

## Testing

With `ENABLE_SPA_LINKS=true`
1. Click through the app and make sure everything works
2. Check the network panel and verify that the page isn't reloaded. (shouldn't see a request for an HTML page)

With `ENABLE_SPA_LINKS=false`
1. Click through the app and make sure everything works
1. Check the network panel and verify that a request for the full page is made
1. Ensure the PDP variant selection does not force a full page load
1. Ensure that the grid filter controls do not force a full page load
